frontend/src/components/Applications/Applications.tsx

```diff
@@ -13,183 +10,235 @@ function wait(ms: number) {
+        }
 
-    function handleAdd() {
-        setAddRotate(true);
+        fetchApplications();
 
-        setTimeout(() => {
-            setAddRotate(false);
-        }, 250);
-        setFormData({
-            applicationId: undefined, // Keine ID, da es sich um eine neue Bewerbung handelt
-            initialData: {
-                companyName: '', // Leeres Feld für Firmenname
-                status: '', // Leeres Feld für Status
-            },
-        });
-        setShowForm(true); // Formular anzeigen
-    }
+        return () => {
+            isMounted = false; // Cleanup beim Unmount
+        };
+    }, [reloadKey, setLoading]);
 
-    function handleToggleDetails(selectedApplication: Application) {
-        setSelectedApplication(selectedApplication);
+    // Debugging-Effekt: Überwacht den Zustand von "selectedApplication"
+    useEffect(() => {
+        console.log("Selected application:", selectedApplication);
+    }, [selectedApplication]); // Führt die Funktion aus, wenn sich "selectedApplication" ändert
+
+    if (loading) {
+        return <LoadingSpinner/>;
     }
 
+
+    const handleToggleDetails = (application: Application) => {
+        if (selectedApplication?.id === application.id) {
+            setSelectedApplication(null); // Detailansicht schließen
+        } else {
+            setSelectedApplication(application); // Detailansicht öffnen
+        }
+    };
+
     function handleEdit(application: Application) {
-        setSelectedApplication(null);
         setFormData({
             applicationId: application.id,
             initialData: {
                 companyName: application.companyName,
                 status: application.status,
+                jobTitle: application.jobTitle,
+                applicationDate: application.applicationDate,
+                jobPostingFoundDate: application.jobPostingFoundDate,
+                applicationEntryCreationDate: application.applicationEntryCreationDate,
+                companyWebsite: application.companyWebsite,
+                companyEmail: application.companyEmail,
+                companyStreet: application.companyStreet,
+                companyHouseNumber: application.companyHouseNumber,
+                phoneNumber: application.phoneNumber,
+                contactPersonFirstName: application.contactPersonFirstName,
+                contactPersonLastName: application.contactPersonLastName,
+                contactPersonEmail: application.contactPersonEmail,
+                jobSource: application.jobSource,
+                jobPostingUrl: application.jobPostingUrl,
+                applicationMethod: application.applicationMethod,
+                applicationPortalUrl: application.applicationPortalUrl,
+                notes: application.notes,
+                uploadedDocuments: application.uploadedDocuments
             },
         });
         setShowForm(true);
+        setSelectedApplication(null);
     }
 
```

Comment:
it's better to build a util for the enum translate & import it, in case u need it somewhere else.

Reply:
Yes, thats perhaps true.

But this was the solution after hours of trouble....because i had a big problem with the "_" and the german umlaute and the Groß-/Kleinschreibung.
This was the best way to avoid more struggle....